zaza/openstack/charm_tests/test_utils.py

```diff
@@ -154,6 +154,36 @@ def setUpClass(cls, application_name=None, model_alias=None):
             model_name=cls.model_name)
         logging.debug('Leader unit is {}'.format(cls.lead_unit))
 
+    def config_current_separate_special_keys(
+            self, special_type_keys, config_keys=None,
+            application_name=None):
+        """Obtain current config and the special type config separately.
+
+        :param config_keys: iterable of strs to index into the current config.
+                            If None, return all keys from the config
+        :type config_keys:  Optional[Iterable[str]]
+        :param special_type_keys: list of keys to be separated out
+        :type special_type_keys: list
+        :param application_name: String application name for use when called
+                                 by a charm under test other than the object's
+                                 application.
+        :type application_name:  Optional[str]
+        :return: Dictionary of previous charm configs applied without the
+                 special type keys provided, and dictionary of the special keys
+                 found in the intended config dictionary.
```

Comment:
I kind of get your meaning (except in the terminology of 'special'.   Please define special in terms of 'things that they do or are'.).

I think it's going to require an example to explain what is going on in the docstring.  i.e. Why should the keys be removed; what's the purpose behind it.